src/components/charts/canvas-d3-radial-chart.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from "react";
+import * as d3 from "d3";
+
+interface IProps {
+  data: number[][];     // (deg,mag) tuples: [[deg,mag], [deg,mag], ...]
+  width: number;
+}
+
+export class CanvasD3RadialChart extends React.Component<IProps> {
+
+  private canvasRef = React.createRef<HTMLCanvasElement>();
+  private svgRef = React.createRef<SVGSVGElement>();
+
+  public componentDidMount() {
+    this.drawChart();
+  }
+
+  public componentDidUpdate() {
+    if (this.canvasRef.current && this.svgRef.current) {
+      // clear everything
+      const ctx = this.canvasRef.current.getContext("2d")!;
+      ctx.clearRect(0, 0, this.props.width, this.props.width);
+      while (this.svgRef.current.lastChild) {
+        this.svgRef.current.removeChild(this.svgRef.current.lastChild);
+      }
+    }
+    this.drawChart();
+  }
+
+  public render() {
+    const width = this.props.width;
+    const relativeStyle: React.CSSProperties = {position: "relative", width, height: width};
+    const absoluteStyle: React.CSSProperties = {position: "absolute", top: 0, left: 0};
+    return (
+      <div style={relativeStyle}>
+        <canvas ref={this.canvasRef} style={absoluteStyle} />
+        <svg ref={this.svgRef} style={absoluteStyle} />
+      </div>
+    );
+  }
+
+  private drawChart() {
+    if (!this.canvasRef.current || !this.svgRef.current) return;
+
+    const { data, width: totalWidth } = this.props;
+
+    const margin = {top: 20, right: 20, bottom: 35, left: 20};
+    const chartWidth = totalWidth - margin.left - margin.right;
+
+    const svgAxes = d3.select(this.svgRef.current)
+      .attr("width", totalWidth)
+      .attr("height", totalWidth)
+      .append("g")
+      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+    const maxMagnitude = d3.max(data, d => d[1]) || 1;
+    const maxRadius = chartWidth / 2;
+    const xScale = d3.scaleLinear()
+      .domain([-maxMagnitude, maxMagnitude])
+      .range([ 0, chartWidth]);
+    const yScale = d3.scaleLinear()
+      .domain([-maxMagnitude, maxMagnitude])
+      .range([ chartWidth, 0]);
+    const magScale = d3.scaleLinear()
+        .domain([0, maxMagnitude])
+        .range([ 0, maxRadius]);
+
+    // add axes. Assume NESW for now, can add props to customize later if needed
+    svgAxes.append("g")
+      .call(d3.axisTop(xScale).tickValues([0]).tickFormat(() => "N"));
+    svgAxes.append("g")
+      .attr("transform", "translate(" + chartWidth + ", 0)")
+      .call(d3.axisRight(yScale).tickValues([0]).tickFormat(() => "W"));
```

Comment:
I think this should probably be "E" for east.

Reply:
:+1: Fixed in the next PR.